js/web/lib/wasm/jsep/webgpu/gpu-data-manager.ts

```diff
@@ -144,15 +152,19 @@ class GpuDataManagerImpl implements GpuDataManager {
 
   // eslint-disable-next-line no-bitwise
   create(size: number, usage = GPUBufferUsage.STORAGE | GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST): GpuData {
-    // !!!
-    // !!! IMPORTANT: TODO: whether we should keep the storage buffer every time, or always create new ones.
-    // !!!                  This need to be figured out by performance test results.
-    // !!!
-
     const bufferSize = calcNormalizedBufferSize(size);
+    if (!this.freeBuffers.has(bufferSize)) {
+      this.freeBuffers.set(bufferSize, []);
+    }
 
-    // create gpu buffer
-    const gpuBuffer = this.backend.device.createBuffer({size: bufferSize, usage});
+    let gpuBuffer;
+    const buffers = this.freeBuffers.get(bufferSize);
+    if (buffers && buffers.length > 0) {
```

Comment:
I think we can always get buffers?

Reply:
perhaps can merge this with the previous lines
```ts
let buffers = this.freeBuffers.get(bufferSize);
if (!buffers) {
  buffers = [];
  this.freeBuffers.set(bufferSize, buffers);
}
if (buffers.length > 0) {
  //...

```
Typescript will be happy with it and this can also reduce the time of accessing the map from 2 to 1 when value does not exist.

Reply:
Done